src/block/TallGrass.php

```diff
@@ -25,14 +25,58 @@
 
 use pocketmine\block\utils\StaticSupportTrait;
 use pocketmine\block\utils\TallGrassTrait;
+use pocketmine\item\Fertilizer;
+use pocketmine\item\Item;
 use pocketmine\math\Facing;
+use pocketmine\math\Vector3;
+use pocketmine\player\Player;
 
 class TallGrass extends Flowable{
 	use TallGrassTrait;
 	use StaticSupportTrait;
 
-	private function canBeSupportedAt(Block $block) : bool{
-		$supportBlock = $block->getSide(Facing::DOWN);
-		return $supportBlock->hasTypeTag(BlockTypeTags::DIRT) || $supportBlock->hasTypeTag(BlockTypeTags::MUD);
+    /** @phpstan-var \Closure() : DoublePlant|null */
+    private ?\Closure $doublePlantVariant;
+
+    /**
+     * @phpstan-param \Closure() : DoublePlant|null $doublePlantVariant
+     */
+    public function __construct(BlockIdentifier $idInfo, string $name, BlockTypeInfo $typeInfo, ?\Closure $doublePlantVariant = null){
+        parent::__construct($idInfo, $name, $typeInfo);
+        $this->doublePlantVariant = $doublePlantVariant;
+    }
+
+    private function canBeSupportedAt(Block $block) : bool{
+        $supportBlock = $block->getSide(Facing::DOWN);
+        return $supportBlock->hasTypeTag(BlockTypeTags::DIRT) || $supportBlock->hasTypeTag(BlockTypeTags::MUD);
+    }
+
+    public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+        if($this->getSide(Facing::UP)->getTypeId() !== BlockTypeIds::AIR){
+            return false;
+        }
+
+		$world = $this->position->getWorld();
+		if($item instanceof Fertilizer){
+			$doubleVariant = $this->getDoublePlantVariant();
+            if($doubleVariant === null){
```

Comment:
Can you please merge this condition with fertilizer check to have `if($item instanceof Fertilizer && ($doubleVariant = $this->getDoublePlantVariant()) !== null){ //bonemeal logic } `. Also, please move getting of the world inside the bonemeal logic.